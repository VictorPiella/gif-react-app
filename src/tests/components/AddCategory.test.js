import React from 'react'
import '@testing-library/jest-dom';

import { shallow } from 'enzyme'
import { AddCategory } from '../../components/AddCategory'

describe('Test <AddCategoriy />', () => {

  const setCategories = jest.fn();
  let wrapper = shallow( <AddCategory setCategories={ setCategories } /> );

  beforeEach( () => {
      jest.clearAllMocks();
      wrapper = shallow( <AddCategory setCategories={ setCategories } /> );

  })
  
  test('should show properlly', () => {

    expect( wrapper ).toMatchSnapshot();
    
  })
  test('should change text box', () => {
    
    const input = wrapper.find('input');
    const value = 'Hello world'
    input.simulate('Change', { target: { value } });

    expect(wrapper.find('p').text().trim() ).toBe( value ); 

  })

  test('should not post data with submit ', () => {
    
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });

    expect( setCategories ).not.toHaveBeenCalled();


  })

  test('should  call setCategories and clean input', () => {

    const value = 'Hello world'
    //simulate inputChange
    wrapper.find('input').simulate('Change', { target: { value } });
    //simulate submit
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });
    //setCategories called
    expect( setCategories ).toHaveBeenCalled();
    expect( setCategories ).toHaveBeenCalledTimes(1);
    expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );
    //inoput empty
    expect(wrapper.find('input').prop('value') ).toBe( '' ); 
    
  })
  
  
  
  
  
})
