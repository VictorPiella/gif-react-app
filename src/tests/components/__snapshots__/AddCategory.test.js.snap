// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AddCategoriy /> should show properlly 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
